### ISSUE TITLE
firewalld module only works with platform-python on oel

### ISSUE THREAD

**Issue by ccamachofg** [ansible-collections/ansible.posix#265]
##### SUMMARY
I am trying to enable some services using the firewalld module:
- name: Enable firewalld services
  firewalld:
    service: "{{item}}"
    permanent: yes
    immediate: yes
    state: enabled
  loop:
  - service1
  - service2
 
This tasks gets executed fine on managed servers where I only have the platform-python. The moment I install a new python (yum install python39) the tasks fails with: Python Module not found: firewalld and its python module are required for this module.
The managed server is using OEL8.4

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
firewalld

##### ANSIBLE VERSION
```
ansible [core 2.11.4]
  config file = /home/jenkins/.ansible.cfg
  configured module search path = ['/home/jenkins/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules', '/home/jenkins/.ansible/collections/ansible_collections/ansible/posix/plugins/modules']
  ansible python module location = /usr/local/lib/python3.8/site-packages/ansible
  ansible collection location = /home/jenkins/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/local/bin/ansible
  python version = 3.8.6 (default, Apr  9 2021, 12:30:30) [GCC 8.4.1 20200928 (Red Hat 8.4.1-1.0.1)]
  jinja version = 3.0.1
  libyaml = True
```

##### COLLECTION VERSION

```
# /home/jenkins/.ansible/collections/ansible_collections
Collection    Version
------------- -------
ansible.posix 1.3.0

# /usr/local/lib/python3.8/site-packages/ansible_collections
Collection    Version
------------- -------
ansible.posix 1.3.0
```

##### CONFIGURATION

```
HOST_KEY_CHECKING(env: ANSIBLE_HOST_KEY_CHECKING) = False
```

##### OS / ENVIRONMENT
Oracle Enterprise Linux 8.4

##### STEPS TO REPRODUCE
from a fresh install of oel8.4 the task works without issues
after I install python39 (sudo yum install python39) it doesn't work anymore

```yaml
- name: Enable firewalld services
  firewalld:
    service: "{{item}}"
    permanent: yes
    immediate: yes
    state: enabled
  loop:
  - service1
  - service2

```

##### EXPECTED RESULTS
The task should execute fine even if not using the platform-python

##### ACTUAL RESULTS

```paste below
failed: [host1] (item=ntp) => {
    "ansible_loop_var": "item",
    "changed": false,
    "invocation": {
        "module_args": {
            "icmp_block": null,
            "icmp_block_inversion": null,
            "immediate": true,
            "interface": null,
            "masquerade": null,
            "offline": null,
            "permanent": true,
            "port": null,
            "port_forward": null,
            "rich_rule": null,
            "service": "ntp",
            "source": null,
            "state": "enabled",
            "target": null,
            "timeout": 0,
            "zone": null
        }
    },
    "item": "ntp",
    "msg": "Python Module not found: firewalld and its python module are required for this module,                         version 0.2.11 or newer required (0.3.9 or newer for offline operations)"
}

```

**Comment by saito-hideki** [comment]
Hi @ccamachofg !
Hi! thank you for reporting this matter. I think this is the expected behavior.
As far as I know, python3-firewalld is included in a part of firewalld packages provided by OS distributors. However, this package(python3-firewalld) is only for platform-python and they are not released for Python 3.8 or 3.9.
In this case, you probably need to set `/usr/libexec/platform-python` as a value of  `ansible_python_interpreter` according to the following document:

- https://docs.ansible.com/ansible/latest/reference_appendices/interpreter_discovery.html#interpreter-discovery
- https://docs.ansible.com/ansible/latest/reference_appendices/config.html#interpreter-python

I hope tihis helps :)

**Comment by saito-hideki** [comment]
Hello @ccamachofg 
Like I mentioned earlier, I think this is not the issue on the firewalld side and the python-firewall library should be provided by OS distributors.
Having said that, I think that it should be described in the module documentation.  So I have added a requirement and descriptions by PR #287.
I will close this issue, but if I misunderstand the situation, please feel free to reopen it.
